module/organisations/helper.js

```diff
@@ -72,18 +71,30 @@ module.exports = class OrganisationsHelper {
                                     });
 
                                 }))
+                                sessionHelpers.set(constants.common.ORGANISATIONS_SESSION,organisationsList);
                             }
                         }
+                      }
 
                     } else if (profileData.result.response.organisations) {
                         let orgList = profileData.result.response.organisations;
                         await Promise.all(orgList.map(async function (orgInfo) {
                             if (roles.includes(constants.common.ORG_ADMIN_ROLE) ||
                                 orgInfo.roles.includes(constants.common.ORG_ADMIN_ROLE)) {
-                                let result = await _getOrganisationDetailsById(orgInfo.organisationId, token);
-                                let orgDetails = { value: orgInfo.organisationId, label: result.orgName };
-
-                                organisationsList.push(orgDetails);
+                                        
+                                    let sessionOrganisationData = sessionHelpers.get(constants.common.ORGANISATIONS_SESSION);
+                                    if(sessionOrganisationData){
+
+                                        let orgData = sessionOrganisationData.filter(orgDetails=>{
+                                            return orgDetails.value == orgInfo.organisationId;
+                                        });
+                                        organisationsList.push(orgData);
+
+                                    }else{
+                                        let result = await _getOrganisationDetailsById(orgInfo.organisationId, token);
+                                        let orgDetails = { value: orgInfo.organisationId, label: result.orgName };
+                                        organisationsList.push(orgDetails);
+                                    }
                             }
```

Comment:
@rakeshSgr - No need of pagination here, and remove the same from function parameters